quantities.hxx

```diff
@@ -47,10 +48,13 @@ auto eta(auto df, const std::string &outputname,
 
 auto phi(auto df, const std::string &outputname,
          const std::string &inputvector) {
-    return df.Define(
-        outputname,
-        [](const ROOT::Math::PtEtaPhiMVector &p4) { return (float)p4.phi(); },
-        {inputvector});
+    return df.Define(outputname,
+                     [](const ROOT::Math::PtEtaPhiMVector &p4) {
+                         if (p4.pt() < 0.0)
```

Comment:
Is pt negative for the default lorentz vector ? --> add a comment

Reply:
also is there any speed difference between using if..else and the conditional operator as we did it in artus ? - if its the same, then i guess we keep if else

Reply:
Yes, I checked that pt is still the same vector as in the initialization. Therefore, I think it is ok to use negative pt values as markers that four-vectors are not properly filled and default values should be used. (Alternatively the length of the corresponding mask or njets could always be checked but that requires to pipe an additional quantity through the functions...)
While defaults can be inserted for eta and pt, the phi value is taken modulo 2pi and therefore the default value must be set manually for retrieving phi.
I don't know whether there are speed differences. I would be surprised if they were measaurably large but we can try to find out...